suites/basic.go

```diff
@@ -0,0 +1,220 @@
+	if err != nil {
+		return fmt.Errorf("Error adding ip family and ip family policy to service spec and upserting: %s", err)
+	}
+
+	publicZone := infra.Zones[0]
+	fmt.Println("About to call Validate Record --------------------------")
+	err = validateRecord(ctx, armdns.RecordTypeAAAA, tests.ResourceGroup, tests.SubscriptionId, *tests.ClusterName, publicZone.GetName(), 4, tests.Service.Status.LoadBalancer.Ingress[0].IP)
+	if err != nil {
+		return fmt.Errorf("%s Record not created in Azure DNS", armdns.RecordTypeAAAA)
+	} else {
+		lgr.Info("finished successfully")
+	}
+
+	return nil
+
+}
+
+var ARecordTest = func(ctx context.Context, infra infra.Provisioned, usePublicZone bool) error {
+	fmt.Printf("%+v\n", infra)
+
+	fmt.Println("In A record test --------------")
+	lgr := logger.FromContext(ctx)
+	lgr.Info("starting test")
+
+	//Currently only provisioning one public and one private zone, no test in this suite tests with more than one of each
+	publicZone := infra.Zones[0]
+	privateZone := infra.PrivateZones[0]
+	clusterName := tests.ClusterName
+
+	//printing values for debug
+	fmt.Println("Infra subId: ", infra.SubscriptionId)
+	fmt.Println("Infra cluster name: ", *clusterName)
+	fmt.Println("Infra rg: ", infra.ResourceGroup.GetName())
+	fmt.Println("Infra zone name: ", publicZone.GetName())
+	fmt.Println("Infra Service name: ", infra.ServiceName)
+
+	err := tests.AnnotateService(ctx, infra.SubscriptionId, *clusterName, infra.ResourceGroup.GetName(), "external-dns.alpha.kubernetes.io/hostname", publicZone.GetName(), infra.ServiceName)
+	if err != nil {
+		lgr.Error("Error annotating service", err)
+		return fmt.Errorf("error: %s", err)
+	}
+
+	var zoneName string
+	if usePublicZone {
+		zoneName = publicZone.GetName()
+	} else {
+		zoneName = privateZone.GetName()
+	}
+	err = validateRecord(ctx, armdns.RecordTypeA, infra.ResourceGroup.GetName(), infra.SubscriptionId, *clusterName, zoneName, 4, tests.Service.Status.LoadBalancer.Ingress[0].IP)
+	if err != nil {
+		return fmt.Errorf("%s Record not created in Azure DNS", armdns.RecordTypeA)
+	} else {
+		lgr.Info("finished successfully")
+	}
+
+	return nil
+}
+
+// Checks to see whether record is created in Azure DNS
+func validateRecord(ctx context.Context, recordType armdns.RecordType, rg, subscriptionId, clusterName, serviceDnsZoneName string, timeout time.Duration, svcIp string) error {
+	fmt.Println()
+	fmt.Println("#0 In validateRecord() function ---------------------")
+
+	lgr := logger.FromContext(ctx)
+	lgr.Info("Checking that Record was created in Azure DNS")
+
+	err := tests.WaitForExternalDns(ctx, timeout, subscriptionId, rg, clusterName)
+	if err != nil {
+		return fmt.Errorf("error waiting for ExternalDNS to start running %w", err)
+	}
+
+	//TODO: change implementation, wait for record creation with poller
+	fmt.Println("Sleeping to wait for record creation ------------")
+	time.Sleep(7 * time.Second)
```

Comment:
Instead of just sleeping for some amount of seconds, you should poll for the records in a loop until either you get a successful result, or the timeout happens.